LEGO1/lego/legoomni/src/race/legoraceactor.cpp

```diff
@@ -1,31 +1,121 @@
 #include "legoraceactor.h"
 
+#include "define.h"
+#include "legocachesoundmanager.h"
+#include "legosoundmanager.h"
+#include "misc.h"
+#include "mxmisc.h"
+#include "mxtimer.h"
+#include "mxvariabletable.h"
+#include "roi/legoroi.h"
+
 DECOMP_SIZE_ASSERT(LegoRaceActor, 0x180)
 
+// Initialized at LEGO1 0x100145a0
+// GLOBAL: LEGO1 0x10102b08
+Mx3DPointFloat LegoRaceActor::g_unk0x10102b08 = Mx3DPointFloat(0.0, 2.0, 0.0);
+
 // FUNCTION: LEGO1 0x100145d0
 LegoRaceActor::LegoRaceActor()
 {
 	m_unk0x70 = 0;
 	m_unk0x08 = 0;
 }
 
-// STUB: LEGO1 0x10014750
-MxS32 LegoRaceActor::VTable0x68(Vector3&, Vector3&, Vector3&)
+// FUNCTION: LEGO1 0x10014750
+// FUNCTION: BETA10 0x100c9bba
+MxS32 LegoRaceActor::VTable0x68(Vector3& p_v1, Vector3& p_v2, Vector3& p_v3)
 {
-	// TODO
-	return 0;
+	MxS32 result = LegoPathActor::VTable0x68(p_v1, p_v2, p_v3);
+
+	if (m_userNavFlag && result) {
+		MxLong time = Timer()->GetTime();
+		if (time - g_unk0x100f3308 > 1000) {
+			g_unk0x100f3308 = time;
+			const char* soundKey = VariableTable()->GetVariable(g_strHIT_ACTOR_SOUND);
+
+			if (soundKey && *soundKey) {
+				SoundManager()->GetCacheSoundManager()->Play(soundKey, NULL, FALSE);
+			}
+		}
+	}
+
+	return result;
 }
 
-// STUB: LEGO1 0x100147f0
-MxU32 LegoRaceActor::VTable0x90(float, Matrix4&)
+// FUNCTION: LEGO1 0x100147f0
+// FUNCTION: BETA10 0x100c9c93
+MxU32 LegoRaceActor::VTable0x90(float p_float, Matrix4& p_transform)
 {
-	// TODO
+	// TODO: Note the similarity to LegoExtraActor::VTable0x90.
+	// Code duplication or is there something deeper going on?
```

Comment:
We should (ideally) understand this part and simplify if possible. If we think it is an actual code duplication, we should adapt the comment.

Reply:
Code duplication is pretty common across the code base and I'm fairly sure in these cases too, they just copy-pasted code and then applied minor adjustments. So I think the comments aren't necessary